client/blocks/payment-request/index.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { getSetting } from '@woocommerce/settings';
+
+/**
+ * Internal dependencies
+ */
+import { PAYMENT_METHOD_NAME } from './constants';
+import { PaymentRequestExpress } from './payment-request-express';
+import { applePayImage } from './apple-pay-preview';
+import { getStripeServerData, loadStripe } from '../stripe-utils';
+
+const ApplePayPreview = () => <img src={ applePayImage } alt="" />;
+
+const canPayStripePromise = loadStripe();
+const componentStripePromise = loadStripe();
+
+let isStripeInitialized = false,
+	canPay = false;
+
+// Initialise stripe API client and determine if payment method can be used
+// in current environment (e.g. geo + shopper has payment settings configured).
+function paymentRequestAvailable( { currencyCode, totalPrice } ) {
+	// Stripe only supports carts of greater value than 30 cents.
+	if ( totalPrice < 30 ) {
+		return false;
+	}
+
+	// If we've already initialised, return the cached results.
+	if ( isStripeInitialized ) {
+		return canPay;
+	}
+
+	return canPayStripePromise.then( ( stripe ) => {
+		if ( stripe === null ) {
```

Comment:
Let's use the established [yoda condition WP standard](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions) even though that's _technically_ only defined for PHP 🙂